src/primitive_core.cc

```diff
@@ -1804,7 +1805,20 @@ PRIMITIVE(process_send) {
 
   unsigned size = 0;
   { MessageEncoder size_encoder(process, null);
-    if (!size_encoder.encode(array)) WRONG_TYPE;
+    if (!size_encoder.encode(array)) {
+      Object* result = null;
+      int class_id = size_encoder.problematic_class_id();
```

Comment:
Can this code be shared? Can the encoder help give the right error if asked politely?